driver-core/src/main/com/mongodb/client/model/expressions/IntegerExpression.java

```diff
@@ -46,5 +46,5 @@ default IntegerExpression subtract(final int subtract) {
     IntegerExpression abs();
 
     <R extends Expression> R passIntegerTo(Function<? super IntegerExpression, ? extends R> f);
-    <R extends Expression> R switchIntegerOn(Function<Branches, ? extends BranchesTerminal<? super IntegerExpression, ? extends R>> on);
+    <R extends Expression> R switchIntegerOn(Function<Branches<IntegerExpression>, ? extends BranchesTerminal<? super IntegerExpression, ? extends R>> on);
```

Comment:
`switchIntegerOn` always passes `Branches<IntegerExpression>` as the input for the `on` function. Therefore, it appears that we'd like to accept `on` functions that take either `Branches<IntegerExpression>` or a supertype of this type, i.e., `Function<? super Branches<IntegerExpression>, ...>`.  This indeed, would have given our users more flexibility:


```java
// if a user has a function like the following
Function<Branches<? extends NumberExpression>, BranchesTerminal<NumberExpression, StringExpression>> branches = null;
// then the user could use it in both of the following pieces of code
of(0).switchNumberOn(f);
of(0).switchIntegerOn(f);
// btw, this code also requires `? super IntegerExpression` in `? extends BranchesTerminal<? super IntegerExpression, ...>`, which we currently have
```

However, similarly to (@stIncMale TODO past the link), it's impossible to obtain `BranchesTerminal<NumberExpression, ...>>` from `Branches<? extends NumberExpression>`, i.e., users can never implement the `branches` function in the above code. Therefore, it appears to be fine to accept just `Function<Branches<IntegerExpression>, ...>` in `switchNumberOn`.